MessageHandler.cpp

```diff
@@ -2,17 +2,85 @@
 
 using namespace BinaryNinja;
 
+const std::set<std::string> arcFunctionNames = {
+    "_objc_retain",
+    "_objc_release",
+    "_objc_autorelease",
+    "_objc_autoreleaseReturnValue",
+    "_objc_retainAutoreleasedReturnValue",
+    "_objc_unsafeClaimAutoreleasedReturnValue"
+};
+
 MessageHandler::MessageHandler(Ref<BinaryView> data)
+    : m_data(data)
 {
+    m_authStubsSection = data->GetSectionByName("__auth_stubs");
+    m_stubsSection = data->GetSectionByName("__stubs");
+
     m_msgSendFunctions = findMsgSendFunctions(data);
+    m_arcFunctions = findARCFunctions(data);
+
+    if (!m_authStubsSection && !m_stubsSection)
+        m_readyForRealAnalysisPass = true;
+    else
+        data->RegisterNotification(this);
```

Comment:
In this function:
* Make stub mutex recursive
* Lock it first thing
* Register notif before calling finder funcs, it will not be able to lock as its called from analysis
* Call find funcs which can grab lock

Once they are complete the SymbolAdded can fire, ensuring there's no stub symbolication race.